pkg/apis/apps/validation/validation_test.go

```diff
@@ -818,3 +818,176 @@ func TestValidateControllerRevisionUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateStatefulSetSpec(t *testing.T) {
```

Comment:
Hi! its not clear to me, how different is this from the `TestValidateStatefulSet` the first test case? could you elaborate on that please?

Reply:
I found there are some cases no covered in ValidateStatefulSetSpec. So I added TestValidateStatefulSetSpec to improve coverage.

Reply:
Could these be added as a part of `TestValidateStatefulSet` to reduce code repetition? also as mentioned before if you could convert the test case to `go sub-test`,  testcases would be more informative.